### ISSUE TITLE
Visible spaces between squares

### ISSUE THREAD

**Issue by TheEyeOfBrows** [CodingTrain/12oclocks#22]
## Example:
![image](https://user-images.githubusercontent.com/8050633/31027684-eefb1f0e-a4ff-11e7-8b43-c51dea19ba07.png)

## Issue:
This affects all clocks whose numbers are drawn using squares (rect).

We're getting visible spaces between squares which are drawn side by side. The source examples have clean flat rectangles, so it would be nice to try and fix this.

### Reference:
![image](https://user-images.githubusercontent.com/8050633/31028023-44adb96a-a501-11e7-9ba3-26d0035d4035.png)

## Aside:
There may be a trick in javascript/p5 to resolve this, maybe to do with anti-aliasing?
I've tried adding a stroke to the shapes, but then we loose the clean corners between shapes.

**Comment by Oripy** [comment]
I have the same issue with clock 3 using arcs. Did not find any reliable workaround.

**Comment by Merijn-DH** [comment]
noSmooth() can be used to disable anti-aliasing, but I tried doing adding this to the clock you mentioned but it had no effect.

**Comment by meiamsome** [comment]
One way you can fix this is to enlarge all of the shapes so that they actually overlap by a small amount so there's no possible gap to show through. The edges all line up nicely still but the corners will look a bit problematic. I guess it'll be a middleground between looking bad one way or another, if there's not a better fix.

You could also look at working out maximal rectangles you can draw to avoid this... 
For instance, the 5 can be drawn as 6 rectangles - where the three bottom ones are obvious and the three top ones overlap one square with the other. Might be a pain to work out how to do that nicely - but there is only 12 characters.

**Comment by Merijn-DH** [comment]
See my solution for clock 5 in my pull request (#36).

**Comment by meiamsome** [comment]
Anyone want to implement this for the final couple of clocks with these issues (8, 12 and probably 7)?

**Comment by Merijn-DH** [comment]
Sure, as soon as I have time!

**Comment by meiamsome** [comment]
Sounds good, I just pushed in changes that let you view singular clocks, so it should be easier to check the clocks now (Click to enlarge)

https://codingtrain.github.io/12oclocks/#clock-07

**Comment by Merijn-DH** [comment]
I fixed 7 and 8. Clock 12 looks good in my browser.. maybe it is different for others?!

**Comment by meiamsome** [comment]
It's very subtle. Here's a picture on my screen:

![image](https://user-images.githubusercontent.com/1187491/31154686-b31ef15a-a8a0-11e7-9a86-55b0a629deb9.png)

This one's harder though, because they're partially transparent.

**Comment by Merijn-DH** [comment]
You're right, I can see in my browser as well. Since it's 2 am for me I'll fix it tomorrow unless someone else would be so kind to look into it :)

**Comment by Merijn-DH** [comment]
Just some observation: the trick used in the last 3 fixes won't work in clock 12 since the fills are transparent and the overlaping would be visible.

**Comment by waielal** [comment]
I might have a solution to this problem though I'm not sure if you'll like it or not. 
I created a second font file that defines for every character the outline shapes.
This way we can use
```
// for each shape
beginShape();
for (...)
    vertex(v[i][0], v[i][1]);
endShape();
```
to draw the character gapless.
What do you think of this idea?

**Comment by meiamsome** [comment]
@waielal I had the same idea - but hadn't got around to implementing it - it should be great for the final clock, and maybe retrofitting some of the others if it's so desired, My idea was to just have the font defined as a function for each of the characters:
```javascript
font2 = {
...
  "3": function(sketch) {
    // beginShape, vertex, endShape etc.
  }
...
}
```

And then use transformations and style changes to position it correctly. It's certainly workable, and would love seeing your take on it

**Comment by meiamsome** [comment]
I think that's all of them - or does 10 need fixing too?

**Comment by waielal** [comment]
10 also has some gaps. But less noticable than the others.

**Comment by meiamsome** [comment]
Fixed Via: #36, #42, #43. #45, #46 
Thanks!
